OCR/frontend/src/components/SortableTable/SortableTable.tsx

```diff
@@ -0,0 +1,124 @@
+                                                          data,
+                                                          sortableBy,
+                                                          defaultSort,
+                                                          defaultDescending = false,
+                                                          columns, formatters = {},
+                                                          columnNames = {},
+                                                      }: SortableTableProps) => {
+    if (!columns) {
+        columns = Object.keys(data[0])
+    }
+    const [sortBy, setSortBy] = useState(defaultSort || columns?.[0])
+    const [isDescending, setIsDescending] = useState(defaultDescending)
+
+    const updateSort = (column: string) => {
+        if (column === sortBy) {
+            setIsDescending(!isDescending)
+        } else {
+            setSortBy(column)
+        }
+    }
+
+    const columnData = data?.[0]?.[sortBy]
+    const columnType = Object.prototype.toString.call(columnData)
+    const sortFunc = SORTS[columnType] || new Intl.Collator(navigator.language).compare
+    const sortedData = data?.toSorted((a, b) => sortFunc(a[sortBy], b[sortBy])) || []
+
+    if (isDescending) {
+        sortedData.reverse()
+    }
+
+
+    return (
+        <>
+            <Table fullWidth striped>
+                <thead>
+                <tr>
+                    {columns.map((c, idx) => {
+                        return <SortableTableHeader key={String(idx)} sortBy={sortBy as string} column={c} name={columnNames?.[c] || c}
+                                                    isDescending={isDescending || false} onClick={updateSort}/>
+                    })}
+                </tr>
+                </thead>
+                <tbody>
+                {sortedData.map((t, idx) => {
+                    return (<tr key={idx}>
+                        {columns?.map((col, colIdx) => {
+                            return <td
+                                key={colIdx}>{formatters?.[col] ? formatters?.[col](t[col]) : t[col]?.toString()}</td>
+                        })}
+                    </tr>)
+                })}
+                </tbody>
+            </Table>
+        </>
+    )
+}
+
+interface SortableTableHeaderProps {
+    sortBy: string,
+    name: string,
+    column: string,
+    isDescending: boolean,
+    onClick: (string) => void,
+    key: string
+}
+
+const SortableTableHeader: FC<SortableTableHeaderProps> = ({
+                                                               sortBy,
+                                                               name, column, isDescending,
+                                                               onClick
+                                                           }: SortableTableHeaderProps) => {
+
+    const isSortedBy = sortBy === column
+    return <>
+        <th onClick={() => onClick(column)}>
```

Comment:
minor nitpick. I believe you remove the fragment here since u have a top level tag with <th>